internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/PartialUpdateTest.java

```diff
@@ -388,6 +388,86 @@ public void testIncrementalPushPartialUpdateClassicFormat() throws IOException {
+      assertCommand(
+          parentControllerClient
+              .createNewStore(storeName, "test_owner", STRING_SCHEMA.toString(), schemaV1.toString()));
+      UpdateStoreQueryParams updateStoreParams =
+          new UpdateStoreQueryParams().setStorageQuotaInByte(Store.UNLIMITED_STORAGE_QUOTA)
+              .setCompressionStrategy(CompressionStrategy.NO_OP)
+              .setActiveActiveReplicationEnabled(true)
+              .setChunkingEnabled(true)
+              .setRmdChunkingEnabled(true)
+              .setHybridRewindSeconds(10L)
+              .setHybridOffsetLagThreshold(2L);
+      ControllerResponse updateStoreResponse =
+          parentControllerClient.retryableRequest(5, c -> c.updateStore(storeName, updateStoreParams));
+      assertFalse(updateStoreResponse.isError(), "Update store got error: " + updateStoreResponse.getError());
+
+      VersionCreationResponse response = parentControllerClient.emptyPush(storeName, "test_push_id", 1000);
+      assertEquals(response.getVersion(), 1);
+      assertFalse(response.isError(), "Empty push to parent colo should succeed");
+      TestUtils.waitForNonDeterministicPushCompletion(
+          Version.composeKafkaTopic(storeName, 1),
+          parentControllerClient,
+          30,
+          TimeUnit.SECONDS);
+
+      assertCommand(parentControllerClient.addValueSchema(storeName, schemaV2.toString()));
+      UpdateStoreQueryParams updateStoreParams2 = new UpdateStoreQueryParams().setWriteComputationEnabled(true);
+      updateStoreResponse =
+          parentControllerClient.retryableRequest(5, c -> c.updateStore(storeName, updateStoreParams2));
+      assertFalse(updateStoreResponse.isError(), "Update store got error: " + updateStoreResponse.getError());
+
+      response = parentControllerClient.emptyPush(storeName, "test_push_id_v2", 1000);
+      assertEquals(response.getVersion(), 2);
+      assertFalse(response.isError(), "Empty push to parent colo should succeed");
+      TestUtils.waitForNonDeterministicPushCompletion(
+          Version.composeKafkaTopic(storeName, 2),
+          parentControllerClient,
+          30,
+          TimeUnit.SECONDS);
+    }
+    VeniceClusterWrapper veniceClusterWrapper = childDatacenters.get(0).getClusters().get(CLUSTER_NAME);
+    SystemProducer veniceProducer = getSamzaProducer(veniceClusterWrapper, storeName, Version.PushType.STREAM);
+    String key = "highway";
+    // Send first value with old schema;
+    GenericRecord record = new GenericData.Record(schemaV1);
+    record.put("id", "101");
+    record.put("name", "U.S. 101");
+    sendStreamingRecord(veniceProducer, storeName, key, record);
+    // Send first value with new schema;
+    GenericRecord recordV2 = new GenericData.Record(schemaV2);
+    recordV2.put("id", "280");
+    recordV2.put("name", "Interstate 280");
+    sendStreamingRecord(veniceProducer, storeName, key, recordV2);
```

Comment:
could you clarify that without the fix which write would fail, with old schema or with new schema, or both?

Reply:
It will failed 2nd PUT, I will add to the java doc line.